rust/src/interaction.rs

```diff
@@ -0,0 +1,106 @@
+use binaryninjacore_sys::*;
+
+use std::ffi::CString;
+use std::path::PathBuf;
+
+use crate::string::BnString;
+
+pub fn get_text_line_input(prompt: &str, title: &str) -> Option<String> {
+    let prompt = CString::new(prompt).unwrap();
+    let title = CString::new(title).unwrap();
+    let mut value: *mut libc::c_char = std::ptr::null_mut();
+
+    let result = unsafe { BNGetTextLineInput(&mut value, prompt.into_raw(), title.into_raw()) };
+    if !result {
+        return None;
+    }
+
+    Some(unsafe { BnString::from_raw(value).to_string() })
```

Comment:
I'd like you more if you did the `... if { None } else { Some(val) } }` pattern.